tests/integration/test_cli_pheno.py

```diff
@@ -711,7 +722,10 @@ def test_overwrite_flag_behaviour(
         + overwrite_flag,
     )
 
-    assert expected_stdout in overwrite_result.output
+    assert (
+        "already exists" in overwrite_result.output
+    ) == output_exists_message_shown
+    assert ("Saved output to" in caplog.text) == output_saved_message_shown
```

Comment:
I wonder if this is maybe too clever for a test. How about more boilerplate to make it super explicit:

```suggestion
was_output_exists_message_shown =  "already exists" in overwrite_result.output
was_output_saved_message_shown = "Saved output to" in caplog.text

assert should_show_output_exists_message == was_output_exists_message_shown
assert should_show_output_saved_message == was_output_saved_message_shown
```